Add discussion of Jupyter Book 2, along with directions to documentation for 2.x. (#2280)
jupyter-book/jupyter-book: README.md

* include computational elements (e.g., code cells) in either type,
* include rich syntax such as citations, cross-references, and numbered equations, and
* using a simple command, run the embedded code cells, [cache](https://jupyter-cache.readthedocs.io/en/latest/) the outputs and convert this content into:
  * a web-based interactive book and
  * a publication-quality PDF.

## Governance of this project and Jupyter Book 2.x

> Jupyter Book 1.x is now in a maintenance phase, where new feature requests have lower priority than bugfixes.

Reviewer: ```suggestion
Jupyter Book 1.x is now in a maintenance phase. We will likely not add new features, and bugfixes will have lower priority than Jupyter Book 2.x efforts.
```

> Jupyter Book 2.x will be built on the new [MyST-MD document engine](https://mystmd.org/). See [this blog post](https://executablebooks.org/en/latest/blog/2024-05-20-jupyter-book-myst/) for more information on our plans, and [this blog post](https://blog.jupyterbook.org/posts/2024-11-15-jupyter-book-2-alpha) about the alpha release of Jupyter Book 2.x.

Reviewer: ```suggestion
Jupyter Book `2.x` is built on the new [MyST-MD document engine](https://mystmd.org/). See [this blog post explaining our plans for 2.0](https://executablebooks.org/en/latest/blog/2024-05-20-jupyter-book-myst/), and [this blog post about the alpha release of 2.x](https://blog.jupyterbook.org/posts/2024-11-15-jupyter-book-2-alpha).
```

> Jupyter Book 2.x is currently being developed on the `next` branch, and documentation for Jupyter Book 2.x can be found at [next.jupyterbook.org](https://next.jupyterbook.org).

Reviewer: ```suggestion
Jupyter Book 2.x is currently being developed on [the `next` branch](https://github.com/jupyter-book/jupyter-book/tree/next), and documentation for Jupyter Book 2.x can be found at [next.jupyterbook.org](https://next.jupyterbook.org).
```
